figures/S5.supplemental_figure5/scripts/S5.r

```diff
@@ -0,0 +1,1448 @@
+        "Correlation",
+        "Granularity",
+        "Intensity",
+        "Location",
+        "Neighbors",
+        "RadialDistribution",
+        "Texture"
+    )
+)
+# reverese the order of the factor levels
+il6_final$feature_group <- factor(
+    il6_final$feature_group,
+    levels = rev(levels(il6_final$feature_group))
+)
+
+il6_final_plot <- plot_coeffs(il6_final, cytokine, shuffle)
+# output
+coef_gg_file <- file.path(paste0(output_path,"/","top_abs_val_coefficients_enet.pdf"))
+# set plot size
+width <- 8.5
+height <- 4
+options(repr.plot.width=width, repr.plot.height=height)
+ggsave(coef_gg_file, il6_final_plot, width=width, height=height, dpi=500)
+il6_final_plot
+
+
+cytokine <- 'CCL4 [NSU]'
+shuffle <- 'final'
+filename <- nested_list$filename[which(nested_list$cytokine == cytokine & nested_list$shuffle == shuffle)]
+# set to string
+filename <- filename[[1]]
+CCL4_final <- read.csv(filename, header = TRUE, sep = ",", stringsAsFactors = FALSE)
+
+CCL4_final <- process_subset_data(CCL4_final)
+head(CCL4_final)
+# factor levels
+# rename the factor levels
+CCL4_final$channel_learned <- factor(
+    CCL4_final$channel_learned,
+    levels = c(
+        "nuclei",
+        "Mito",
+        "ER",
+        "gasdermin",
+        "PM",
+        "other"
+    ),
+    labels = c(
+        "Nuclei",
+        "Mito",
+        "ER",
+        "GasderminD",
+        "AGP",
+        "Other"
+    )
+)
+CCL4_final$feature_group <- factor(
+    CCL4_final$feature_group,
+    levels = c(
+        "AreaShape",
+        "Correlation",
+        "Granularity",
+        "Intensity",
+        "Location",
+        "Neighbors",
+        "RadialDistribution",
+        "Texture"
+    )
+)
+# reverese the order of the factor levels
+CCL4_final$feature_group <- factor(
+    CCL4_final$feature_group,
+    levels = rev(levels(CCL4_final$feature_group))
+)
+
+CCL4_final_plot <- plot_coeffs(CCL4_final, cytokine, shuffle)
+# output
+coef_gg_file <- file.path(paste0(output_path,"/","top_abs_val_coefficients_enet.pdf"))
+# set plot size
+width <- 8.5
+height <- 4
+options(repr.plot.width=width, repr.plot.height=height)
+ggsave(coef_gg_file, CCL4_final_plot, width=width, height=height, dpi=500)
+CCL4_final_plot
+
+
+# set cell type
+cell_type <- "PBMC"
+
+
+# set path for data of all models
+data_path <- file.path(paste0("../../../6.bulk_Morphology_Elastic_Network/4.model_performance/results/regression/", cell_type, "/", "all_model_performance.csv"))
+df <- read.csv(data_path)
+# setfigure path
+figure_path <- file.path(paste0("../figures/regression/", cell_type, "/"))
+# make the directory if it doesn't exist
+dir.create(figure_path, recursive = TRUE, showWarnings = FALSE)
+
+
+# fix the col name
+df <- df %>%
+  mutate(secreted_proteins = case_when(
+    secreted_proteins == "MMP-1 [NSU]" ~ "MMP-1",
+    secreted_proteins == "VEGFR-1 [NSU]" ~ "VEGFR-1",
+    secreted_proteins == "CCL4 [NSU]" ~ "CCL4",
+    secreted_proteins == "MMP-12 [NSU]" ~ "MMP-12",
```

Comment:
It seems like there may be an easier way to remove a suffix from all columns, although I am not able to understand this code: https://stackoverflow.com/questions/45960269/removing-suffix-from-column-names-using-rename-all

Reply:
Yeah, I have to remove suffixes and unique strings from each though so it changes